src/main/java/games/strategy/engine/framework/GameDataManager.java

```diff
@@ -140,63 +137,14 @@ private static GameData loadGameInSerializationFormat(
-        // so, what we do here is try to see if our installed copy of triplea includes older jars with it that are the
-        // same engine as was
-        // used for this savegame, and if so try to run it
-        try {
-          final String newClassPath = GameRunner.findOldJar(readVersion, true);
-          // ask user if we really want to do this?
-          final String messageString = "<html>This TripleA engine is version " + ClientContext.engineVersion()
-              + " and you are trying to open a savegame made with version " + readVersion.toString()
-              + "<br>However, this TripleA cannot open any savegame made by any engine other than engines with the "
-              + "same first three version numbers as it (x_x_x_x)."
-              + "<br><br>TripleA now comes with older engines included with it, and has found the engine to run this "
-              + "savegame. This is a new feature and is in 'beta' stage."
-              + "<br>It will attempt to run a new instance of TripleA using the older engine jar file, and this "
-              + "instance will only be able to play this savegame."
-              + "<br><b>You may choose to either Close or Keep the current instance of TripleA!</b> (If hosting, you "
-              + "must close it). Please report any bugs or issues."
-              + "<br><br>Do you wish to continue?</html>";
-          final String yesClose = "Yes & Close Current";
-          final String yesOpen = "Yes & Do Not Close";
-          final String cancel = "Cancel";
-          final Object[] options = new Object[] {yesClose, yesOpen, cancel};
-          final JOptionPane pane = new JOptionPane(messageString, JOptionPane.PLAIN_MESSAGE,
-              JOptionPane.YES_NO_CANCEL_OPTION, null, options, yesClose);
-          final JDialog window = pane.createDialog(null, "Run old jar to open old Save Game?");
-          window.setVisible(true);
-          final Object buttonPressed = pane.getValue();
-          if (buttonPressed == null || buttonPressed.equals(cancel)) {
-            return null;
-          }
-          final boolean closeCurrentInstance = buttonPressed.equals(yesClose);
-          GameRunner.startGame(savegamePath, newClassPath);
-          if (closeCurrentInstance) {
-            ThreadUtil.sleep(1000);
-            System.exit(0);
-          }
-        } catch (final IOException e) {
-          if (ClientFileSystemHelper.areWeOldExtraJar()) {
-            throw new IOException("<html>Please run the default TripleA and try to open this game again. "
-                + "<br>This TripleA engine is old and kept only for backwards compatibility and can only open "
-                + "savegames created by engines with these first 3 version digits: "
-                + ClientContext.engineVersion().toStringFull("_", true) + "</html>");
-          } else {
-            throw new IOException(error);
-          }
-        }
-        return null;
+        throw new IOException(error);
```

Comment:
The exception does not seem right, seems like "control flow via exception handling". Users having different versions, or loading a different version, is not an error case, it's a likely scenario that users can trigger. It may be impractical to do the control flow more properly, in which case a TODO may be best.

Reply:
That's what I thought too, but I wanted to keep the old behaviour in order to not accidentally break something 🙈 
It seems I did a mistake, by not correctly reverse-engineering the present code... Will fix that!

Reply:
@DanVanAtta this turns out to be neccessary, reverted my change